### DISCUSSION TITLE
Resize adjacent windows

### DISCUSSION THREAD

**Discussion by newsonaut** [rxhanson/RectanglePro-Community#378] [Q&A] [answered]
In the Settings pane, I checked "Resize adjacent windows after dragging window edge" and in the Window Throw pane, I checked "Resize adjacent windows".

My experience is that this works a couple of times then stops. Am I doing something wrong?

Thanks for any help!

**Comment by rxhanson** [comment]
The window edges need to be pretty close to be picked up. One thing that can cause this is if you resize a window past the smallest size of another window, and the window edges are no longer aligned. Is this what you're experiencing?

**Comment by newsonaut** [comment]
Example: Make one Finder window take up the left half, the other take up the right half. There is a gutter of a few pixels in between. In this case it doesn't work at all.
If I nudge them together so there is no gap, it works but only a few times — the number varies. Sometimes, it will work if I try again, sometimes not.
I'm afraid I can't find a pattern that would be more useful for trouble-shooting.

**Reply by rxhanson** [reply]
Thanks for the additional information. The other factor here is that there is a lot of computation that Rectangle Pro has to do to determine which window edges to respect. Sometimes, the combination of time taken to process this information along with the speed at which you perform your actions and the applications that are running can affect the performance of this functionality. It also could be something that is more prevalent on some Macs than others. Hopefully, I can figure out an optimization that works around these limitations.

**Comment by newsonaut** [comment]
You might want to have a look at 1Piece. It's the only other Mac app I know of that has this functionality, although it's bundled with a confusing array of other functions.

**Reply by rxhanson** [reply]
Thanks for the heads up. I'll see what I can do.

**Comment by rxhanson** [comment, marked as answer]
Please try v3.0.15 (update from within the app or at https://rectangleapp.com/pro/versions). I think it should be working properly now.

**Reply by newsonaut** [reply]
That's much better — thanks for working on this. It's pretty much the only thing I like about Windows.
